lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -0,0 +1,231 @@
+require 'active_merchant/billing/gateways/cecabank/cecabank_common'
+
+module ActiveMerchant
+  module Billing
+    class CecabankJsonGateway < Gateway
+      include CecabankCommon
+
+      CECA_ACTIONS_DICTIONARY = {
+        purchase: :REST_AUTORIZACION,
+        authorize: :REST_PREAUTORIZACION,
+        capture: :REST_COBRO_PREAUTORIZACION,
+        refund: :REST_DEVOLUCION,
+        void: :REST_ANULACION
+      }.freeze
+
+      CECA_REASON_TYPES = {
+        installment: :I,
+        recurring: :R,
+        unscheduled: :C
+      }.freeze
+
+      CECA_INITIATOR = {
+        merchant: :N,
+        cardholder: :S
+      }.freeze
+
+      self.test_url = 'https://tpv.ceca.es/tpvweb/rest/procesos/'
+      self.live_url = 'https://pgw.ceca.es/tpvweb/rest/procesos/'
+
+      def authorize(money, creditcard, options = {})
+        build_purchase(:authorize, money, creditcard, options)
+      end
+
+      def capture(money, identification, options = {})
+        authorization, operation_number, _network_transaction_id = identification.split('#')
+
+        post = {}
+        options[:operation_number] = operation_number
+        add_auth_invoice_data(:capture, post, money, authorization, options)
+
+        commit('compra', post)
+      end
+
+      def purchase(money, creditcard, options = {})
+        build_purchase(:purchase, money, creditcard, options)
+      end
+
+      def void(identification, options = {})
+        authorization, operation_number, money, _network_transaction_id = identification.split('#')
+        options[:operation_number] = operation_number
+        build_cancellation(:void, money.to_i, authorization, options)
+      end
+
+      def refund(money, identification, options = {})
+        authorization, operation_number, _money, _network_transaction_id = identification.split('#')
+        options[:operation_number] = operation_number
+        build_cancellation(:refund, money, authorization, options)
+      end
+
+      private
+
+      def build_purchase(action, money, creditcard, options)
+        post = { parametros: { accion: CECA_ACTIONS_DICTIONARY[action] } }
+
+        add_invoice(post, money, options)
+        add_creditcard(post, creditcard)
+        add_stored_credentials(post, creditcard, options)
+        add_three_d_secure(post, options)
+
+        commit('compra', post)
+      end
+
+      def build_cancellation(action, money, authorization, options = {})
```

Comment:
💬 Style/Naming:

Same observation this method is not just `building` the cancellation